kernel/kernel-api/src/main/java/io/delta/kernel/statistics/DataFileStatistics.java

```diff
@@ -0,0 +1,261 @@
+  }
+
+  /**
+   * Get the number of records in the data file.
+   *
+   * @return Number of records in the data file.
+   */
+  public long getNumRecords() {
+    return numRecords;
+  }
+
+  /**
+   * Get the minimum values of the columns in the data file. The map may contain statistics for only
+   * a subset of columns in the data file.
+   *
+   * @return Map of column to minimum value of it in the data file.
+   */
+  public Map<Column, Literal> getMinValues() {
+    return minValues;
+  }
+
+  /**
+   * Get the maximum values of the columns in the data file. The map may contain statistics for only
+   * a subset of columns in the data file.
+   *
+   * @return Map of column to minimum value of it in the data file.
+   */
+  public Map<Column, Literal> getMaxValues() {
+    return maxValues;
+  }
+
+  /**
+   * Get the number of nulls of columns in the data file. The map may contain statistics for only a
+   * subset of columns in the data file.
+   *
+   * @return Map of column to number of nulls in the data file.
+   */
+  public Map<Column, Long> getNullCounts() {
+    return nullCounts;
+  }
+
+  public String serializeAsJson() {
+    return JsonUtil.generate(
+        gen -> {
+          gen.writeStartObject();
+          gen.writeNumberField("numRecords", numRecords);
+
+          if (dataSchema != null) {
+            gen.writeObjectFieldStart("minValues");
+
+            writeJsonValues(
+                gen,
+                dataSchema,
+                minValues,
+                new Column(new String[0]),
+                (g, v) -> writeJsonValue(g, v));
+            gen.writeEndObject();
+
+            gen.writeObjectFieldStart("maxValues");
+            writeJsonValues(
+                gen,
+                dataSchema,
+                maxValues,
+                new Column(new String[0]),
+                (g, v) -> writeJsonValue(g, v));
+            gen.writeEndObject();
+
+            gen.writeObjectFieldStart("nullCounts");
+            writeJsonValues(
+                gen, dataSchema, nullCounts, new Column(new String[0]), (g, v) -> g.writeNumber(v));
+            gen.writeEndObject();
+          }
+
+          gen.writeEndObject();
+        });
+  }
+
+  private <T> void writeJsonValues(
+      JsonGenerator generator,
+      StructType schema,
+      Map<Column, T> values,
+      Column parentColPath,
+      JsonUtil.JsonValueWriter<T> writer)
+      throws IOException {
+    if (schema == null) {
+      return;
+    }
+    for (StructField field : schema.fields()) {
+      Column colPath = parentColPath.append(field.getName());
+      if (field.getDataType() instanceof StructType) {
+        generator.writeObjectFieldStart(field.getName());
+        writeJsonValues(generator, (StructType) field.getDataType(), values, colPath, writer);
+        generator.writeEndObject();
+      } else {
+        T value = values.get(colPath);
+        if (value != null) {
```

Comment:
we may want to type check the Literal matches the data type to make sure to not write incorrect stats